src/VizPOIAnalytics/index.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import DeckGL from 'deck.gl';
+import { StaticMap } from 'react-map-gl';
+import POIAnalyticsControlPanel from './POIAnalyticsControlPanel';
+import POIAnalyticsRenderLayers from './POIAnalyticsRenderLayers';
+import Tooltip from './components/Tooltip';
+import * as CONSTANTS from './utils/constants';
+import lightingEffect from './utils/lightingEffects';
+import {
+  getValInUSD,
+  getInitialControlPanelSettings,
+  getTooltipFormattedDetails,
+  getBoundingBoxDetailsFromCurrentViewport,
+  fetchPOIDetailsFromFOAMAPI,
+} from './utils/helper';
+
+class VizPOIAnalytics extends React.Component {
+  constructor(props) {
+    super(props);
+    this.fetchAllPOIDetailsInCurrentViewport = this.fetchAllPOIDetailsInCurrentViewport.bind(this);
+    this.state = {
+      viewport: CONSTANTS.INITIAL_VIEWPORT,
+      hover: {
+        x: 0,
+        y: 0,
+        hoveredObject: null,
+      },
+      points: [],
+      FOAMTokenInUSD: 0,
+      settings: getInitialControlPanelSettings(CONSTANTS.HEXAGON_CONTROLS),
+      userResponse: false,
+    };
+  }
+
+  async componentDidMount() {
+    this.setState({
+      FOAMTokenInUSD: await getValInUSD(),
+    });
+    this.setUserLocation();
+  }
```

Comment:
Instead of setting state twice once for FOAMTokenInUSD and then setting user location,

1. fetch FOAM's value in USD
2. get userLocation from setUserLocation function (modified to getUserLocation)
3. When you have got approriate responses from 1 and 2, setState together.

Reply:
Check this code: https://es6console.com/k2rici3t/